app/gui/view/component-browser/searcher-list-panel/src/navigator.rs

```diff
@@ -0,0 +1,255 @@
+        frp::extend! { network
+            eval params.strong_color((&c)
+                icon.borrow().as_ref().map(|icon| icon.strong_color.set(c.into()))
+            );
+            eval params.weak_color((&c)
+                icon.borrow().as_ref().map(|icon| icon.weak_color.set(c.into()))
+            );
+        }
+
+        Self { display_object, logger, icon, icon_id, params: params.clone_ref() }
+    }
+
+    fn update(&self, model: &Self::Model) {
+        if !self.icon_id.get().contains(model) {
+            let size = Vector2(icon::SIZE, icon::SIZE);
+            let icon = model.create_shape(&self.logger, size);
+            icon.strong_color.set(self.params.strong_color.value().into());
+            icon.weak_color.set(self.params.weak_color.value().into());
+            self.display_object.add_child(&icon);
+            *self.icon.borrow_mut() = Some(icon);
+            self.icon_id.set(Some(*model));
+        }
+    }
+
+    fn set_max_width(&self, _max_width_px: f32) {}
+
+    fn set_label_layer(&self, _label_layer: &Layer) {}
+}
+
+
+// === IconParams ===
+
+#[derive(Clone, CloneRef, Debug)]
+pub struct Params {
+    pub strong_color: frp::Sampler<color::Rgba>,
+    pub weak_color:   frp::Sampler<color::Rgba>,
+}
+
+impl Default for Params {
+    fn default() -> Self {
+        let network = frp::Network::new("searcher_list_panel::navigator::Params::default");
+        frp::extend! { network
+            default_color <- source::<color::Rgba>().sampler();
+        }
+        Self { strong_color: default_color.clone_ref(), weak_color: default_color.clone_ref() }
+    }
+}
+
+
+
+// =================
+// === Navigator ===
+// =================
+
+
+// === Section enum ===
+
+/// Three sections of the Searcher List Panel. See the
+/// [Component Browser Design Document](https://github.com/enso-org/design/blob/e6cffec2dd6d16688164f04a4ef0d9dff998c3e7/epics/component-browser/design.md).
+#[derive(Debug, Copy, Clone)]
+pub enum Section {
+    SubModules,
+    LocalScope,
+    Favourites,
+}
+
+impl Default for Section {
+    fn default() -> Self {
+        Section::Favourites
+    }
+}
+
+
+// === Navigator ===
+
+/// A section navigator bar. Contains two sets of buttons placed on the left of the Searcher List
+/// Panel.
+///
+/// The first set on top of the bar contains "Libraries" and "Marketplace" buttons. The second
+/// set on the bottom contains section navigation buttons used to quickly scroll to a specific
+/// section.
+#[derive(Debug, Clone, CloneRef)]
+pub struct Navigator {
+    display_object:     display::object::Instance,
+    network:            frp::Network,
+    bottom_buttons:     list_view::ListView<Icon>,
+    top_buttons:        list_view::ListView<Icon>,
+    pub chosen_section: frp::Source<Option<Section>>,
+}
+
+const TOP_BUTTONS: [icon::Id; 2] = [icon::Id::Libraries, icon::Id::Marketplace];
+const BOTTOM_BUTTONS: [icon::Id; 3] = [icon::Id::SubModules, icon::Id::LocalScope, icon::Id::Star];
```

Comment:
We've been chatting that these buttons should be placed in an auto-layout. I dont see an auto-layout implementation as part of this PR neither explanation that it will be added later. CC @farmaazon

Reply:
> We've been chatting that these buttons should be placed in an auto-layout. I dont see an auto-layout implementation as part of this PR neither explanation that it will be added later. CC @farmaazon

We decided to skip layouters for now, because our timeline is very tight.